amplitude.js

```diff
@@ -270,15 +253,130 @@ Amplitude.prototype.runQueuedFunctions = function () {
   this._q = []; // clear function queue after running
 };
 
-Amplitude.prototype._loadSavedUnsentEvents = function(unsentKey, queue) {
-  var savedUnsentEventsString = localStorage.getItem(unsentKey);
+/**
+ * Set an item in local storage. Decorate item key with the api key
+ * to avoid collisions.
+ *
+ * @property item: name of the item to store
+ * @property value: the value to store
+ */
+Amplitude.prototype.setLocalStorage = function(item, value) {
+  var key = LocalStorageKeys[item] + '_' + this.options.apiKey.slice(0, 6);
```

Comment:
why do we need this extra level of indirection of `LocalStorageKeys[item]`? why not just pass in `LocalStorageKeys.item` into this function and not rely on passing string keys?